test/test_nestedtensor.py

```diff
@@ -8558,6 +8549,33 @@ def f(*args, **kwargs):
 
                     self.assertEqualNoncontigAware(grads_compile, grads_ref)
 
+    @torch._dynamo.config.patch(capture_dynamic_output_shape_ops=True)
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
```

Comment:
just to clarify: this flag is required for things to work? This is because the index_put implementation requires access to the offsets data, right?

Reply:
Right. Index_select/Index_put operations are doing split underneath that needs access to offsets.item(), which only works with `capture_scalar_outputs=True`